### PR TITLE
Burst average product improvements

### PR DESCRIPTION
**Pull request by lbesnard**
@ggalibert @mhidas 

intend to fix
https://github.com/aodn/data-services/issues/713
https://github.com/aodn/content/issues/284
https://github.com/aodn/data-services/issues/557

### CODE REVIEW

File: ANMN/burst_averaged_product/burst_average.py [hunk 1]
```diff
@@ -255,8 +255,12 @@ def create_burst_average_netcdf(input_netcdf_file_path, output_dir):
     output_netcdf_obj.date_created = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
 
     depth_burst_mean_val = burst_vars['DEPTH']['var_mean']
-    output_netcdf_obj.geospatial_vertical_min = (np.ma.masked_array(depth_burst_mean_val, np.isnan(depth_burst_mean_val))).min()
-    output_netcdf_obj.geospatial_vertical_max = (np.ma.masked_array(depth_burst_mean_val, np.isnan(depth_burst_mean_val))).max()
+    if (np.ma.masked_array(depth_burst_mean_val, np.isnan(depth_burst_mean_val))).mask.all():
```
Comment (mhidas) [inline ANMN/burst_averaged_product/burst_average.py]: Wouldn't this be much clearer and have exactly the same effect?
```python
if np.isnan(depth_burst_mean_val).all():
```

File: ANMN/burst_averaged_product/burst_average.py [hunk 2]
```diff
@@ -255,8 +255,12 @@ def create_burst_average_netcdf(input_netcdf_file_path, output_dir):
...
+        output_netcdf_obj.geospatial_vertical_min = np.double(input_netcdf_obj['NOMINAL_DEPTH'][:])
+        output_netcdf_obj.geospatial_vertical_max = np.double(input_netcdf_obj['NOMINAL_DEPTH'][:])
+    else:
+        output_netcdf_obj.geospatial_vertical_min = (np.ma.masked_array(depth_burst_mean_val, np.isnan(depth_burst_mean_val))).min()
+        output_netcdf_obj.geospatial_vertical_max = (np.ma.masked_array(depth_burst_mean_val, np.isnan(depth_burst_mean_val))).max()
```
Comment (mhidas) [inline ANMN/burst_averaged_product/burst_average.py]: Again, no real need to convert to a masked_array just to get min & max (especially not twice!) You can just say
```python
output_netcdf_obj.geospatial_vertical_min = np.nanmin(depth_burst_mean_val)
output_netcdf_obj.geospatial_vertical_max = np.nanmax(depth_burst_mean_val)
```

File: ANMN/burst_averaged_product/burst_average.py [hunk 3]
```diff
@@ -346,7 +346,8 @@ def create_burst_average_netcdf(input_netcdf_file_path, output_dir):
 
     # append original gatt to burst average gatt
     gatt = 'comment'
-    setattr(output_netcdf_obj, gatt, ('%s. %s' % (getattr(input_netcdf_obj, gatt, ''), getattr(output_netcdf_obj, gatt) )).lstrip('. '))
+    if getattr(input_netcdf_obj, gatt, '') != '':
```
[the remark this answers is not in the document]
Comment (lbesnard) [reply ANMN/burst_averaged_product/burst_average.py]: I wanted to make sure that the FV01 file would not have a empty string for the attribute value. But I guess the FV01 should pass the checker anyway

**Comment by lbesnard** [comment]
thanks for the review, will do the mod !

**Comment by lbesnard** [comment]
@mhidas done

**Comment by lbesnard** [comment]
this is probably ready to be merged

**Comment by mhidas** [comment]
@lbesnard Thanks. I'll take a look next week.

**Pull request merged**